optimum/habana/transformers/training_args.py

```diff
@@ -610,8 +628,9 @@ def _setup_devices(self) -> "torch.device":
 
             if self.use_lazy_mode:
                 logger.info("Enabled lazy mode.")
-            else:
-                os.environ["PT_HPU_LAZY_MODE"] = "2"
+            elif not self.torch_compile:
+                if os.getenv("PT_HPU_LAZY_MODE", "1") != "0":
+                    os.environ["PT_HPU_LAZY_MODE"] = "2"
```

Comment:
If I understood correctly, this will have to be removed when SynapseAI 1.13 is released as eager mode won't be supported anymore right?

Reply:
yes.